.devcontainer/launch.sh

```diff
@@ -11,17 +11,38 @@ print_help() {
     echo "the top-level devcontainer in .devcontainer/devcontainer.json will be used."
     echo ""
     echo "Options:"
-    echo "  -c, --cuda       Specify the CUDA version. E.g., 12.2"
-    echo "  -H, --host       Specify the host compiler. E.g., gcc12"
-    echo "  -d, --docker     Launch the development environment in Docker directly without using VSCode."
-    echo "  -h, --help       Display this help message and exit."
+    echo "  -c, --cuda               Specify the CUDA version. E.g., 12.2"
+    echo "  -H, --host               Specify the host compiler. E.g., gcc12"
+    echo "  -d, --docker             Launch the development environment in Docker directly without using VSCode."
+    echo "  --gpus gpu-request       GPU devices to add to the container ('all' to pass all GPUs)."
+    echo "  -e, --env list           Set additional container environment variables."
+    echo "  -v, --volume list        Bind mount a volume."
+    echo "  -h, --help               Display this help message and exit."
+}
+
+_upvar() {
+    if unset -v "$1"; then
+        if (( $# == 2 )); then
+            eval $1=\"\$2\";
+        else
+            eval $1=\(\"\${@:2}\"\);
+        fi;
+    fi
 }
 
 parse_options() {
-    local OPTIONS=c:H:dh
-    local LONG_OPTIONS=cuda:,host:,docker,help
-    local PARSED_OPTIONS=$(getopt -n "$0" -o "${OPTIONS}" --long "${LONG_OPTIONS}" -- "$@")
+    local -;
+    set -euo pipefail;
+
+    local UNPARSED="${!#}";
+    set -- "${@:1:$#-1}";
```

Comment:
It's not clear what this is doing, can you add a comment?

Reply:
Done